web-ui/src/components/feedback_submit_form/FeedbackSubmitForm.jsx

```diff
@@ -148,23 +205,116 @@ const FeedbackSubmitForm = ({ requesteeName, requestId, request }) => {
     })
   }
 
+  const onSliderChange = (event, index, marks, value) => {
+    let questionAnswerCopy = [...questionAnswerPairs];
+    questionAnswerCopy[index].answer.answer = marks?.find((mark) => mark.value === value)?.text;
+    currentlyBeingEdited = index;
+    setQuestionAnswerPairs(questionAnswerCopy);
+    updateAnswer();
+  }
+
+  const onRadioChange = (event, index, value) => {
+    let questionAnswerCopy = [...questionAnswerPairs];
+    questionAnswerCopy[index].answer.answer = value;
+    currentlyBeingEdited = index;
+    setQuestionAnswerPairs(questionAnswerCopy);
+    updateAnswer()
+  }
 
   const onChangeHandler = (event, index) => {
-    let questionAnswerCopy = questionAnswerPairs;
+    let questionAnswerCopy = [...questionAnswerPairs];
     questionAnswerCopy[index].answer.answer = event.target.value;
     currentlyBeingEdited = index;
     setQuestionAnswerPairs(questionAnswerCopy);
     updateAnswer();
   }
 
+  const getQuestionHeader = (index, isReview) => isReview && index === 1
+    && (<h2>How often has the team member has displayed each of the following in the past year, if at all...</h2>);
```

Comment:
2nd `has` needs to be removed but this is just a non-breaking typo.

Reply:
Good catch, thanks!